components/keyboard-shortcuts/index.js

```diff
@@ -13,7 +14,19 @@ class KeyboardShortcuts extends Component {
 	componentWillMount() {
 		this.mousetrap = new Mousetrap;
 		forEach( this.props.shortcuts, ( callback, key ) => {
-			this.mousetrap.bind( key, callback );
+			// Normalize callback, which can be passed as either a function or
+			// an array of [ callback, isGlobal ]
+			let isGlobal = false;
+			if ( Array.isArray( callback ) ) {
+				isGlobal = callback[ 1 ];
+				callback = callback[ 0 ];
+			}
+
+			if ( isGlobal ) {
+				this.mousetrap.bindGlobal( key, callback );
+			} else {
+				this.mousetrap.bind( key, callback );
+			}
 		} );
```

Comment:
Do you think the shortcuts prop could change over time? Should we implement `componentWillReceivePorps` or not?

Reply:
>Do you think the shortcuts prop could change over time? Should we implement `componentWillReceivePorps` or not?

Potentially, though the effect could be imitated by rendering separate `KeyboardShortcuts` components (or changing the `key` of an existing one). Doesn't seem like a need we currently have, but then again, could be a potential source of confusion for future usage.

Reply:
I started looking at this, and realized that there may be some non-trivial checks to consider whether keyboard bindings need to be updated.

https://gist.github.com/aduth/52f5c4ff561723c2d71b7e9d7fc142e7

Instead, and since there's currently no need, I improved documentation to note that shortcuts changes won't take effect and a suggested workaround (26016802582015e096a0d88a9a959c6ee8fec552).